xbmc/Application.cpp

```diff
@@ -4085,16 +4085,24 @@ PlayBackRet CApplication::PlayFile(const CFileItem& item, bool bRestart)
       CJobManager::GetInstance().PauseJobs();
     }
 
-    // don't hold graphicscontext here since player
-    // may wait on another thread, that requires gfx
-    CSingleExit ex(g_graphicsContext);
+    if (item.Exists()) // this is also an implicit call to CWakeOnAccess::WakeUpHost() if enabled
```

Comment:
I'm not 100% sure if we should still hold the graphics lock when calling `Exists()` because it can take several seconds (depending on the WoL timeout) and holding the graphics lock for that long is never good especially if you want to show a "working..." dialog.